include/envoy/tracing/http_tracer.h

```diff
@@ -31,13 +31,15 @@ class Config {
   virtual const std::vector<Http::LowerCaseString>& requestHeadersForTags() const PURE;
 };
 
-/*
- * Basic abstraction for span.
- */
 class Span;
 
 typedef std::unique_ptr<Span> SpanPtr;
 
+class SpanFinalizer;
```

Comment:
you can just do full class declaration here and move everything from https://github.com/lyft/envoy/pull/1029/files#diff-7b5f0d3c634c8c1ac2d4fcaa26541dd9R79